File: luci-app-campus-auth/htdocs/luci-static/resources/view/campus-auth.js
'use strict';
'require view';
'require form';
'require fs';
'require ui';

return view.extend({
	render: function() {
		var m = new form.Map('campus-auth', '校园网认证');
		var s = m.section(form.NamedSection, 'main', 'campus_auth', '认证设置');
		var o;

		s.addremove = false;

		o = s.option(form.Flag, 'enabled', '启用认证');
		o.default = '1';
		o.rmempty = false;

		o = s.option(form.Value, 'student_id', '学号');
		o.rmempty = false;

		o = s.option(form.Value, 'password', '密码');
		o.password = true;
		o.rmempty = false;

		o = s.option(form.ListValue, 'login_method', '运营商');
		o.value('1', '电信');
		o.value('2', '移动');
		o.value('3', '联通');
		o.value('4', '校园网');
		o.default = '1';

		o = s.option(form.Value, 'interface', '认证网络接口', '用于读取当前 IPv4 地址，通常为 wan。');
		o.default = 'wan';
		o.rmempty = false;

		o = s.option(form.Flag, 'schedule_enabled', '启用每日检查');
		o.default = '1';
		o.rmempty = false;

		o = s.option(form.Value, 'schedule_hour', '检查小时');
		o.datatype = 'range(0,23)';
		o.default = '3';
		o.rmempty = false;

		o = s.option(form.Value, 'schedule_minute', '检查分钟');
		o.datatype = 'range(0,59)';
		o.default = '0';
		o.rmempty = false;

		return m.render();
	},

	handleSaveApply: function(ev) {
		return this.handleSave(ev).then(function() {
			return ui.changes.apply();
		}).then(function() {
			return fs.exec('/etc/init.d/campus-auth', [ 'reload' ]);
		}).then(function(result) {
			if (result.code !== 0)
				throw new Error('Failed to reload campus-auth: ' + result.stderr);
		});
	}
});

File: luci-app-campus-auth/root/usr/share/luci/model/cbi/campus-auth.lua
local sys = require "luci.sys"

m = Map("campus-auth", "校园网认证", "配置校园网 Portal 认证和每日自动检查。")
m.on_after_commit = function(self)
    sys.call("/etc/init.d/campus-auth reload >/dev/null 2>&1")
end

s = m:section(NamedSection, "main", "campus_auth", "认证设置")
s.addremove = false
s.anonymous = true

enabled = s:option(Flag, "enabled", "启用认证")
enabled.default = 1
enabled.rmempty = false

student = s:option(Value, "student_id", "学号")
student.datatype = "string"
student.rmempty = false

password = s:option(Value, "password", "密码")
password.password = true
password.rmempty = false

method = s:option(ListValue, "login_method", "运营商")
method:value("1", "电信")
method:value("2", "移动")
method:value("3", "联通")
method:value("4", "校园网")
method.default = "1"

iface = s:option(Value, "interface", "认证网络接口", "用于读取当前 DHCP IPv4 地址，通常为 wan。")
iface.default = "wan"
iface.rmempty = false

scheduled = s:option(Flag, "schedule_enabled", "启用每日检查")
scheduled.default = 1
scheduled.rmempty = false

hour = s:option(Value, "schedule_hour", "检查小时")
hour.datatype = "range(0,23)"
hour.default = "3"
hour.rmempty = false

minute = s:option(Value, "schedule_minute", "检查分钟")
minute.datatype = "range(0,59)"
minute.default = "0"
minute.rmempty = false

return m
